sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/KeyAsyncClient.java

```diff
@@ -694,8 +1127,30 @@
     @ServiceMethod(returns = ReturnType.COLLECTION)
     public PagedFlux<DeletedKey> listDeletedKeys() {
         return new PagedFlux<>(
-            () -> monoContext(context -> listDeletedKeysFirstPage()),
-            continuationToken -> monoContext(context -> listDeletedKeysNextPage(continuationToken)));
+            () -> withContext(context -> listDeletedKeysFirstPage(context)),
+            continuationToken -> withContext(context -> listDeletedKeysNextPage(continuationToken, context)));
```

Comment:
It seems unfortunate that this method is basically a duplicate of the package-private listDeletedKeys below. It feels like this could be made less repetitive by getting the Azure-core Context and passing that into the method below.

Reply:
As discussed, will keep it as is for now and filed an issue to revisit this later - https://github.com/Azure/azure-sdk-for-java/issues/4464